File: omnes.Web/Modules/Parametros/TiposDomicilio/TiposDomicilioRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace omnes.Parametros;

[ConnectionKey("Default"), Module("Parametros"), TableName("TiposDomicilio")]
[DisplayName("Tipos Domicilio"), InstanceName("Tipos Domicilio")]
[ReadPermission("SuperAdmin")]
[ModifyPermission("SuperAdmin")]
public sealed partial class TiposDomicilioRow : Row<TiposDomicilioRow.RowFields>, IIdRow, INameRow
{
    class RowTemplate
    {
        [DisplayName("Id Tipo Domicilio"), Identity, IdProperty]
        public int? IdTipoDomicilio { get; set; }

        [DisplayName("Nombre Tipo Domicilio"), Size(100), NotNull, QuickSearch, NameProperty]
        public string NombreTipoDomicilio { get; set; }
    }
}

File: omnes.Web/Modules/ServerTypes/Parametros/TiposDomicilioRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface TiposDomicilioRow {
    IdTipoDomicilio?: number;
    NombreTipoDomicilio?: string;
}

export abstract class TiposDomicilioRow {
    static readonly idProperty = 'IdTipoDomicilio';
    static readonly nameProperty = 'NombreTipoDomicilio';
    static readonly localTextPrefix = 'Parametros.TiposDomicilio';
    static readonly deletePermission = 'SuperAdmin';
    static readonly insertPermission = 'SuperAdmin';
    static readonly readPermission = 'SuperAdmin';
    static readonly updatePermission = 'SuperAdmin';

    static readonly Fields = fieldsProxy<TiposDomicilioRow>();
}